src/middlewared/middlewared/etc_files/local/smb4_share.conf

```diff
@@ -64,6 +68,32 @@
                 if share['hostsdeny']:
                     pc[share["name"]].update({"hosts deny": share['hostsdeny']})
 
+                """
+                   If we detect that this share is also an AFP share, then enable cross-protocol
+                   locking, and change the way alternate datastream naming syntax so that it matches
+                   what netatalk does.
+
+                   With NFS, turn off oplocks since FreeBSD doesn't currently support kernel oplocks
+                   and force strict locking.
+                """
+                if any(filter(lambda x: x['path'] in share['path'], db['afp_shares'])):
```

Comment:
I am thinking we may need to be a bit more strict for finding shares.

There are two situations here that may need to be account here.

1. Say afp share has /mnt/tank/foo and smb share has /mnt/tank/foobar, it will match but they are different shares.

2. Say afp share has /mnt/tank/foo and smb share has /mnt/tank/foo/bar and each of them is a different dataset.

I think 1 is a must, 2 depends if the shares can cross filesystem boundaries, I know NFS doesn't, do others?

Thoughts?